src/mscorlib/shared/System/BitConverter.cs

```diff
@@ -298,6 +476,13 @@ public static bool ToBoolean(byte[] value, int startIndex)
             return value[startIndex] != 0;
         }
 
+        public static bool ToBoolean(ReadOnlySpan<byte> value)
+        {
+            if (value.Length < sizeof(byte))
```

Comment:
We're actually writing out a byte (the 1 or 0 below based on whether the bool is true or false).  And since we're writing a byte below, it also makes sense to check against the sizeof(byte).